### PR TITLE
Update cpu context for vcpu runtime context and cpu S3

### PR DESCRIPTION
**Pull request by JasonChenCJ**

### CODE REVIEW

**Comment by lijinxia** [comment]
OK to verify

File: hypervisor/arch/x86/guest/vmsr.c [hunk 1]
```diff
@@ -162,7 +161,7 @@ int rdmsr_vmexit_handler(struct vcpu *vcpu)
 	case MSR_IA32_TIME_STAMP_COUNTER:
 	{
 		/* Add the TSC_offset to host TSC to get guest TSC */
-		v = rdtsc() + vcpu->arch_vcpu.contexts[cur_context].tsc_offset;
+		v = rdtsc() + exec_vmread(VMX_TSC_OFFSET_FULL);
```
[the remark this answers is not in the document]
Comment (JasonChenCJ) [reply hypervisor/arch/x86/guest/vmsr.c]: Thanks, will update all.

File: hypervisor/arch/x86/cpu.c [hunk 2]
```diff
@@ -430,6 +430,18 @@ void bsp_boot_init(void)
 	ASSERT(offsetof(struct run_context, ia32_spec_ctrl) ==
 		CPU_CONTEXT_OFFSET_IA32_SPEC_CTRL,
 		"run_context ia32_spec_ctrl offset not match");
+	ASSERT(offsetof(struct run_context, rflags) ==
```
Comment (dongyaozu) [inline hypervisor/arch/x86/cpu.c]: Per discussion, we want to move those OFFSET check to a separate file/API which is not needed for link...

**Comment by lijinxia** [comment]
Verified+1

**Pull request merged**